### ISSUE TITLE
Release 0.7.2

### ISSUE THREAD

**Issue by johannes-b** [keptn/keptn#2431]
# Keptn 0.7.2

* **Link to milestone: [Milestone 0.7.2](https://github.com/keptn/keptn/milestone/21)**
* **Release Date: <= October XX th, 2020**
* **Release Manager: Christian Kreuzberger @christian-kreuzberger-dtx**

### Step 1 - Release of keptn/spec [OPTIONAL] 

**Artifact:** GitHub Release `keptn/spec:0.1.6` (also referred as ``spec.keptn.sh/0.1.6``)

**Assignee**: @bacherfl 

**Repo**: https://github.com/keptn/spec

**Note**: Not every Keptn release requires a new Keptn spec.

- [x] All release-related PRs/discussions are resolved and are merged into master (please check back with the Issue/PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] *Draft* a new release https://github.com/keptn/spec/releases with release notes (collect them from PR/discussion titles) on master branch
- [x] Ask for feedback (e.g., on Slack)
- [x] **ONLY FOR RELEASE MANAGER**: Release notes approved
- [x] If you're happy with everything, pull the trigger and release the spec
- [x] **ONLY FOR RELEASE MANAGER**: keptn/spec:x.y.z released on GitHub
- [x] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)

---
### Step 2 - Release of keptn/examples

**Artifact:** GitHub Release `keptn/examples:0.7.2`

**Assignee**: @bacherfl 

**Repo**: https://github.com/keptn/examples

**Note**: Use the same version as for keptn/keptn (every Keptn release requires a new release of Keptn examples)

- [x] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Created `release-x.x.x` branch: `git checkout -b release-x.x.x`
- [x] _on release-x.x.x branch:_ releasenotes_develop.md renamed to releasenotes_**Vx.x.x**.md (**AUTOMATION WANTED**)
- [x] _on release-x.x.x branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [x] _on release-x.x.x branch:_ Update README.md with links to *Maintained Examples and Keptn Use cases* (not for beta releases)
- [x] _on release-x.x.x branch:_ `git commit`
- [x] Create PR from release-x.x.x to master and request reviews (review on release notes is mandatory)
- [x] **ONLY FOR RELEASE MANAGER**: Release notes approved
- [x] Once the PR is approved and merged into master, *draft* a new release https://github.com/keptn/examples/releases with the release notes on the `release-x.x.x` branch (**AUTOMATION WANTED**)
- [x] If you're happy with everything, pull the trigger and release the examples
- [x] **ONLY FOR RELEASE MANAGER**: keptn/examples:x.x.x released on GitHub
- [x] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [x] Created new and empty release notes (releasenotes_develop.md) file on master branch

--- 
### Step 3 - Release of Keptn kubernetes-utils (Note: Not needed this time, as there were no changes since the last release)

**Artifact:** GitHub Release of `keptn/kubernetes-utils:vx.x.x` 

**Assignee**: Firstname Lastname (Insert Github Handle here)

**Repo**: https://github.com/keptn/kubernetes-utils

**Note:** The `v` in the version, which can be used in GoLang using `go get github.com/keptn/kubernetes-utils@vx.x.x`

- [ ] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [ ] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [ ] Created `release-x.x.x` branch: `git checkout -b release-x.x.x`
- [ ] _on release-x.x.x branch:_ version bump (**Is this needed?**)
- [ ] _on release-x.x.x branch:_ releasenotes_develop.md renamed to releasenotes_**Vx.x.x**.md
- [ ] _on release-x.x.x branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [ ] _on release-x.x.x branch:_ `git commit`
- [ ] Create PR from release-x.x.x to master and request reviews (review on release notes is mandatory)
- [ ] **ONLY FOR RELEASE MANAGER**: Release notes approved
- [ ] Once the PR is approved and merged into master, *draft* a new release https://github.com/keptn/kubernetes-utils/releases with the release notes on the `release-x.x.x` branch - please note the prefix `v` in the version name (**AUTOMATION WANTED**)
- [ ] If you're happy with everything, pull the trigger and release the examples
- [ ] **ONLY FOR RELEASE MANAGER**: keptn/kubernetes-utils:vx.x.x released on GitHub
- [ ] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [ ] Created new and empty release notes (releasenotes_develop.md) file on master branch

--- 
### Step 4 - Release of Keptn go-utils

**Assignee:** @bacherfl 

**Artifact:** GitHub Release of `keptn/go-utils:v0.7.2` 

**Note:** The `v` in the version, which can be used in GoLang using `go get github.com/keptn/go-utils@vx.x.x`

- [x] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Created `release-0.7.2` branch: `git checkout -b release-x.x.x`
- [x] _on release-0.7.2 branch:_ version bump (**Is this needed?**)
- [x] _on release-0.7.2 branch:_ releasenotes_develop.md renamed to releasenotes_**Vx.x.x**.md
- [x] _on release-0.7.2 branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [x] _on release-0.7.2 branch:_ `git commit`
- [x] Create PR from release-0.7.2 to master and request reviews (review on release notes is mandatory)
- [x] **ONLY FOR RELEASE MANAGER**: Release notes approved
- [x] Once the PR is approved and merged into master, *draft* a new release https://github.com/keptn/go-utils/releases with the release notes on the `release-0.7.2` branch - please note the prefix `v` in the version name (**AUTOMATION WANTED**)
- [x] If you're happy with everything, pull the trigger and release the examples
- [x] **ONLY FOR RELEASE MANAGER**: keptn/go-utils:v0.7.2 released on GitHub
- [x] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [x] Created new and empty release notes (releasenotes_develop.md) file on master branch

## Release of Keptn

:warning: **WARNING/STOP/DANGERZONE**: Do not start with this section unless **all the above steps** have been completed successfully.

---
### Step 5 - Release of keptn-contrib services

**Assignee:** @bacherfl 

**Please update version tags of below services accordingly; use semantic versioning**

**foreach:**
- [x] dynatrace-service > **dynatrace-service:0.10.0**
- [x] dynatrace-sli-service > **dynatrace-sli-service:0.7.0**
- [x] prometheus-service > **prometheus-service:0.3.6**
- [x] prometheus-sli-service > **prometheus-sli-service:0.2.3**
- [x] argo-service > **argo-service:0.1.2**
- [x] servicenow-service > **servicenow-service:0.2.1**
- [x] notification-service > **notification-service:0.3.1**
- [ ] ~~unleash-service > **unleash-service:0.2.0**~~

follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 1

---
### Step 6 - Release of Keptn
- [x] Upgrade command (`keptn upgrade`) ready (from last version to current one)
- [x] Upgrade docs available
- [x] Git submodule for *specification* directory in keptn release branch are updated (e.g., `cd specification && git fetch && git checkout a.b.c && cd .. && git add specification && git commit -m "updated spec to a.b.c" && git push`) 
   - **Note**: In case the `specification` directory in the keptn/keptn repo is empty, execute the following command **before**: `git submodule update --init`.
- [x] All feature/bugfix/hotfix branches are merged into master
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Latest go-utils version is used (run: `go get github.com/keptn/go-utils@v0.7.2`) in all sub-folders that use it (see [this code snippet](https://github.com/keptn/go-utils/blob/247b98ef868c6234cf5cb5f852551c4746d472ea/.travis.yml#L56-L69) to use an example)
- [x] Latest kubernetes-utils version is used (run: `go get github.com/keptn/kubernetes-utils@vx.x.x`) in all sub-folders that use it: `api`, `cli`, `configuration-service`, `gatekeeper-service`, `helm-service`.
- [ ] @johannes-b Keptn release notes are up-to-date and spec version is referenced in release notes 
- [x] releasenotes_develop.md renamed to releasenotes_**Vx.x.x**.md
- [x] Created `release-x.x.x` branch: `git checkout -b release-x.x.x`
- [x] _on release-x.x.x branch:_ Set release version in: `installer/manifests/keptn/Chart.yaml` 
- [x] _on release-x.x.x branch:_ Set release version in: `installer/manifests/keptn/charts/continuous-delivery/Chart.yaml` 
- [x] _on release-x.x.x branch:_ Set release version in: `installer/manifests/keptn/charts/continuous-delivery/charts/openshift/Chart.yaml` 
- [x] _on release-x.x.x branch:_ Set release version in: `installer/manifests/keptn/charts/control-plane/Chart.yaml` 
- [x] _on release-x.x.x branch:_ Set release version in: `installer/manifests/keptn/charts/control-plane/charts/mongodb/Chart.yaml` 
- [x] Changes pushed to release branch: `git push` > Travis built image > see [DockerHub](https://cloud.docker.com/u/keptn/repository/list) & see [GCP Cloud Storage](https://console.cloud.google.com/storage/browser/keptn-cli?project=sai-research)
- [ ] Ask stakeholders to review the release notes (*Note:* Do not use a GitHub DRAFT Release, it does not allow editing from multiple parties at the same time and we will lose changes**)

Check on DockerHub: 
- [x] api:x.x.x
- [x] bridge2:x.x.x
- [x] configuration-service:x.x.x
- [x] distributor:x.x.x
- [x] eventbroker-go:x.x.x
- [x] gatekeeper-service:x.x.x
- [x] helm-service:x.x.x
- [x] jmeter-service:x.x.x
- [x] lighthouse-service:x.x.x
- [x] mongodb-datastore:x.x.x
- [x] openshift-route-service:x.x.x
- [x] shipyard-service:x.x.x
- [x] remediation-service:x.x.x

Check on GCP Cloud Storage: 
- [x] keptn-cli x.x.x
- [x] keptn-installer x.x.x

Notify certain users and stakeholders about the release, such that they can test it and ensure compatibility with their services/docs. Make sure to provide instructions on how to download the CLI and install Keptn. **Make sure to let them know that this might not the final release, rather that it is a release candidate.**

- [x] Notify https://github.com/keptn/tutorials CODEOWNERS @christian-kreuzberger-dtx 
- [x] Notify https://github.com/keptn-contrib repo owners @christian-kreuzberger-dtx 
- [x] Notify https://github.com/keptn-sandbox repo owners @christian-kreuzberger-dtx 

---
### Step 7 - Preparation phase

- [ ] keptn-sockshop/carts released **@christian-kreuzberger-dtx NOT NEEDED**
- [x] keptn.sh/docs/integrations/ is up-to-date
- [x] keptn CLI command docu generated and updated on keptn.sh/docs/x.x.x/reference/cli/commands/
- [x] keptn.sh/docs/ is up-to-date
- [ ] tutorials.keptn.sh: update of tutorials in this branch: xyz

---
### Step 8 - Test phase

Upgrader:
- [x] Upgrade command tested for full installation (GKE or K3s)
- [x] Upgrade command tested for quality-gates/control-plane installation (GKE or K3s)
- [ ] Upgrade command tested for full installation (OpenShift 3.11 or MiniShift)
- [ ] Upgrade command tested for quality-gates/control-plane installation (OpenShift 3.11 or MiniShift)

Tests:
- Download of **keptn CLI x.x.x** > [ToDo: LINK to CLI]()
- Install keptn cli and execute: `keptn version`
- For keptn-contrib services: follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 2
- Pre-view of **keptn.sh/docs** > [ToDo: LINK]()
- [ ] GCP - GKE - **Keptn core team**
  - [x] installation: `keptn install`
    - [x] quality gates UC completed according to keptn.sh/docs with Dynatrace
    - [ ] quality gates UC completed according to keptn.sh/docs with Prometheus
  - [x] @bacherfl Installation: `keptn install --use-case=continuous-delivery`
    - [x] All tutorials completed with Dynatrace
        - [x] Onboarding a Service
        - [x] Deployments with Quality Gates
        - [x] Up-scale with Dynatrace
        - [x] Self-healing with Feature Flags
    - [x] All tutorials completed with Prometheus 
        - [x] Onboarding a Service
        - [x] Deployments with Quality Gates
        - [x] Up-scale with Prometheus
  - [x] Bridge verified
  - [x] Delete a project used in the use cases: `keptn delete project`
  - [x] Uninstall Keptn from cluster: `keptn uninstall`
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version
- [ ] Azure - AKS
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version
- [x] @bacherfl AWS - EKS
  - [x] All tutorials completed with Dynatrace
      - [ ] Onboarding a Service
      - [ ] Deployments with Quality Gates
      - [ ] Up-scale with Dynatrace
      - [ ] Self-healing with Feature Flags
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version
- [ ] (AG) OpenShift 3.11 - MiniShift
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used OpenShift version
- [ ] OpenShift 4.x
  - [ ] All tutorials completed with Dynatrace
      - [ ] Onboarding a Service
      - [ ] Deployments with Quality Gates
      - [ ] Self-healing with Feature Flags
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used OpenShift version
  - [ ] Update support matrix on keptn.sh/docs and release notes (key announcement + special thanks to Marc)
- [x] Kubernetes (e.g., Rancher)
  - [x] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version

---
### Step 9 - Official release of keptn-contrib services

**foreach:**
- [x] @bacherfl dynatrace-service > **dynatrace-service:0.10.0**
- [x] @bacherfl dynatrace-sli-service > **dynatrace-sli-service:0.7.0**
- [x] @johannes-b prometheus-service > **prometheus-service:0.3.6**
- [x] @johannes-b prometheus-sli-service > **prometheus-sli-service:0.2.3**
- [x] @johannes-b argo-service > **argo-service:0.1.2**
- [x] @bacherfl servicenow-service > **servicenow-service::0.2.1**
- [x] @bacherfl notification-service > **notification-service:0.3.1**
- [ ] ~~unleash-service > **unleash-service:0.2.0**~~

follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 3

---
### Step 10 - Official release of Keptn

**Note**: Some steps are only for GA releases, not for alpha, beta, etc.
- [ ] PR from release-x.x.x to master (**only if needed**) **@christian-kreuzberger-dtx not needed**
- [x] Travis built image > see [DockerHub](https://cloud.docker.com/u/keptn/repository/list) @christian-kreuzberger-dtx 

Check on DockerHub: @christian-kreuzberger-dtx 
- [x] api:x.x.x
- [x] bridge2:x.x.x
- [x] configuration-service:x.x.x
- [x] distributor:x.x.x
- [x] eventbroker-go:x.x.x
- [x] gatekeeper-service:x.x.x
- [x] helm-service:x.x.x
- [x] jmeter-service:x.x.x
- [x] lighthouse-service:x.x.x
- [x] mongodb-datastore:x.x.x
- [x] openshift-route-service:x.x.x
- [x] remediation-service:x.x.x
- [x] shipyard-service:x.x.x

Check on GCP Cloud Storage:  @christian-kreuzberger-dtx 
- [x] keptn-cli x.x.x
- [x] keptn-installer x.x.x

Release Keptn
- [x] Keptn on GitHub released based on release notes (**AUTOMATION WANTED**) @christian-kreuzberger-dtx
- [x] Make sure CLI binaries are added as assets to the release (**AUTOMATION WANTED**)
- [x] **release-x.x.x** branch tagged with **x.x.x** (**AUTOMATION WANTED**) @christian-kreuzberger-dtx
- [ ] Merge PR from **release-x.x.x** to **master** (**merge only if needed**) @christian-kreuzberger-dtx  **incompatible release, we cherry-picked the respective changes to master**
- [x] Created new and empty release notes (releasenotes_develop.md) file on master branch @christian-kreuzberger-dtx

---
### Final steps
- [x] Update **Quick Start** on keptn.sh with newest release (**only for GA, not for alpha/beta releases**)
- [x] Update [get.keptn.sh repo](https://github.com/keptn/get.keptn.sh) (which auto-uploads to get.keptnsh S3 bucket with) newest release (**only for GA, not for alpha/beta releases**) @christian-kreuzberger-dtx
- [x] Add release notes to: https://keptn.sh/docs/news/release_announcements/ (**AUTOMATION WANTED**)
- [ ] Release of keptn.sh
- [x] Send out Slack/Twitter notification
- [ ] Send out Release-Email
- [ ] Make sure any (relevant) changes to the release checklist are also saved in the [wiki](https://github.com/keptn/keptn/wiki/Release-Checklist) @christian-kreuzberger-dtx
